From b27a455795951c2a909131c368c902f6205a19af Mon Sep 17 00:00:00 2001
From: Motomu Utsumi <motomuman@google.com>
Date: Mon, 20 Feb 2023 16:39:22 +0900
Subject: [PATCH 1/1] cronet: Address AutoBoxing API linter issues

Suppress them since the API return values are @Nullable

Bug: 265674359
Test: TH
Change-Id: I205151096c70251f9fba6a8595a42bb98caec74c
---
 .../net/http/ConnectionMigrationOptions.java  |  9 ++++---
 .../api/src/android/net/http/DnsOptions.java  | 24 ++++++++++++-------
 .../api/src/android/net/http/QuicOptions.java |  3 ++-
 3 files changed, 24 insertions(+), 12 deletions(-)

diff --git a/components/cronet/android/api/src/android/net/http/ConnectionMigrationOptions.java b/components/cronet/android/api/src/android/net/http/ConnectionMigrationOptions.java
index b93f87577..65602ec75 100644
--- a/components/cronet/android/api/src/android/net/http/ConnectionMigrationOptions.java
+++ b/components/cronet/android/api/src/android/net/http/ConnectionMigrationOptions.java
@@ -53,7 +53,8 @@ public class ConnectionMigrationOptions {
     /**
      * See {@link Builder#setEnableDefaultNetworkMigration}
      */
-    @Nullable
+    // SuppressLint since return value is @Nullable
+    @Nullable @SuppressLint("AutoBoxing")
     public Boolean getEnableDefaultNetworkMigration() {
         return mEnableDefaultNetworkMigration;
     }
@@ -61,7 +62,8 @@ public class ConnectionMigrationOptions {
     /**
      * See {@link Builder#setEnablePathDegradationMigration}
      */
-    @Nullable
+    // SuppressLint since return value is @Nullable
+    @Nullable @SuppressLint("AutoBoxing")
     public Boolean getEnablePathDegradationMigration() {
         return mEnablePathDegradationMigration;
     }
@@ -103,7 +105,8 @@ public class ConnectionMigrationOptions {
      * See {@link Builder#setAllowNonDefaultNetworkUsage}
      */
     @Experimental
-    @Nullable
+    // SuppressLint since return value is @Nullable
+    @Nullable @SuppressLint("AutoBoxing")
     public Boolean getAllowNonDefaultNetworkUsage() {
         return mAllowNonDefaultNetworkUsage;
     }
diff --git a/components/cronet/android/api/src/android/net/http/DnsOptions.java b/components/cronet/android/api/src/android/net/http/DnsOptions.java
index de800ab6d..1c97bbeb4 100644
--- a/components/cronet/android/api/src/android/net/http/DnsOptions.java
+++ b/components/cronet/android/api/src/android/net/http/DnsOptions.java
@@ -66,7 +66,8 @@ public final class DnsOptions {
     /**
      * See {@link Builder#setUseHttpStackDnsResolver}
      */
-    @Nullable
+    // SuppressLint since return value is @Nullable
+    @Nullable @SuppressLint("AutoBoxing")
     public Boolean getUseHttpStackDnsResolver() {
         return mUseHttpStackDnsResolver;
     }
@@ -74,7 +75,8 @@ public final class DnsOptions {
     /**
      * See {@link Builder#setPersistHostCache}
      */
-    @Nullable
+    // SuppressLint since return value is @Nullable
+    @Nullable @SuppressLint("AutoBoxing")
     public Boolean getPersistHostCache() {
         return mPersistHostCache;
     }
@@ -82,8 +84,9 @@ public final class DnsOptions {
     /**
      * See {@link Builder#setEnableStaleDns}
      */
+    // SuppressLint since return value is @Nullable
     @Experimental
-    @Nullable
+    @Nullable @SuppressLint("AutoBoxing")
     public Boolean getEnableStaleDns() {
         return mEnableStaleDns;
     }
@@ -99,8 +102,9 @@ public final class DnsOptions {
     /**
      * See {@link Builder#setPreestablishConnectionsToStaleDnsResults}
      */
+    // SuppressLint since return value is @Nullable
     @Experimental
-    @Nullable
+    @Nullable @SuppressLint("AutoBoxing")
     public Boolean getPreestablishConnectionsToStaleDnsResults() {
         return mPreestablishConnectionsToStaleDnsResults;
     }
@@ -145,22 +149,26 @@ public final class DnsOptions {
     // SuppressLint to be consistent with other cronet code
     @Experimental @SuppressLint("UserHandleName")
     public static class StaleDnsOptions {
-        @Nullable
+        // SuppressLint since return value is @Nullable
+        @Nullable @SuppressLint("AutoBoxing")
         public Long getFreshLookupTimeoutMillis() {
             return mFreshLookupTimeoutMillis;
         }
 
-        @Nullable
+        // SuppressLint since return value is @Nullable
+        @Nullable @SuppressLint("AutoBoxing")
         public Long getMaxExpiredDelayMillis() {
             return mMaxExpiredDelayMillis;
         }
 
-        @Nullable
+        // SuppressLint since return value is @Nullable
+        @Nullable @SuppressLint("AutoBoxing")
         public Boolean getAllowCrossNetworkUsage() {
             return mAllowCrossNetworkUsage;
         }
 
-        @Nullable
+        // SuppressLint since return value is @Nullable
+        @Nullable @SuppressLint("AutoBoxing")
         public Boolean getUseStaleOnNameNotResolved() {
             return mUseStaleOnNameNotResolved;
         }
diff --git a/components/cronet/android/api/src/android/net/http/QuicOptions.java b/components/cronet/android/api/src/android/net/http/QuicOptions.java
index bea375744..b71c0a209 100644
--- a/components/cronet/android/api/src/android/net/http/QuicOptions.java
+++ b/components/cronet/android/api/src/android/net/http/QuicOptions.java
@@ -130,7 +130,8 @@ public class QuicOptions {
     /**
      * See {@link Builder#setInMemoryServerConfigsCacheSize}
      */
-    @Nullable
+    // SuppressLint since return value is @Nullable
+    @Nullable @SuppressLint("AutoBoxing")
     public Integer getInMemoryServerConfigsCacheSize() {
         return mInMemoryServerConfigsCacheSize;
     }
-- 
2.40.0.rc1.284.g88254d51c5-goog

